include/dem/particle_particle_contact_force.h

```diff
@@ -1129,131 +1045,300 @@ class ParticleParticleContactForce
-          particle_one_type, particle_two_type)] *
-        radius_times_overlap_sqrt +
-      DBL_MIN;
-    double tangential_damping_constant =
-      normal_damping_constant * sqrt(model_parameter_st / model_parameter_sn);
-
-    // Calculation of normal force. The cohesive term is computed here.
-    const double cohesive_term =
-      2. * M_PI * effective_radius *
-      this->effective_surface_energy[vec_particle_type_index(
-        particle_one_type, particle_two_type)];
-    double normal_force_norm =
-      normal_spring_constant * normal_overlap +
-      normal_damping_constant * normal_relative_velocity_value;
-
-    // Calculation of tangential force. Since we need damping tangential force
-    // in the gross sliding again, we define it as a separate variable
-    Tensor<1, 3> damping_tangential_force =
-      tangential_damping_constant * tangential_relative_velocity;
-    tangential_force =
-      (tangential_spring_constant * contact_info.tangential_overlap) +
-      damping_tangential_force;
+    const double F_so = 6.283185307179586 * // 2. * M_PI
+                        this->effective_radius *
+                        this->effective_surface_energy.at(property_index);
 
-    double coulomb_threshold =
-      this->effective_coefficient_of_friction[vec_particle_type_index(
-        particle_one_type, particle_two_type)] *
-      normal_force_norm;
+    const double delta_0 = -std::sqrt(
+      this->effective_hamaker_constant.at(property_index) / (6. * F_so));
 
-    // Check for gross sliding
-    const double tangential_force_norm = tangential_force.norm();
-    if (tangential_force_norm > coulomb_threshold)
+    // Cohesive force. This will need to be added to the
+    // first vector inside the tuple.
+    double cohesive_term;
+    if (normal_overlap > delta_0)
       {
-        // Gross sliding occurs and the tangential overlap and tangential
-        // force are limited to Coulomb's criterion
-        contact_info.tangential_overlap =
-          (coulomb_threshold *
-             (tangential_force / (tangential_force_norm + DBL_MIN)) -
-           damping_tangential_force) /
-          (tangential_spring_constant + DBL_MIN);
+        cohesive_term = -F_so;
+
+        if (normal_overlap > 0.)
+          {
+            this->template calculate_hertz_mindlin_limit_overlap_contact<
+              Parameters::Lagrangian::ParticleParticleContactForceModel::DMT>(
+              contact_info,
+              tangential_relative_velocity,
+              normal_relative_velocity_value,
+              normal_unit_vector,
+              normal_overlap,
+              particle_one_properties,
+              particle_two_properties,
+              normal_force,
+              tangential_force,
+              particle_one_tangential_torque,
+              particle_two_tangential_torque,
+              rolling_resistance_torque);
+          }
+        else
+          {
+            contact_info.tangential_overlap = Tensor<1, 3>();
```

Comment:
won't the tangential contact always be set to zero anyway if the overlap between the particles is larger than zero?

Reply:
Larger than 0 means a contact between particles